src/nectarchain/trr_test_suite/linearity.py

```diff
@@ -134,13 +136,27 @@ def main():
     index = 0
     for run in runlist:
         print("PROCESSING RUN {}".format(run))
+        pedestal_tool = PedestalNectarCAMCalibrationTool(
+            progress_bar=True,
+            run_number=run,
+            max_events=12000,
+            events_per_slice=5000,
+            log_level=20,
+            output_path=output_dir + f"/pedestal_{run}.h5",
+            overwrite=True,
+            filter_method=None,
+            method="FullWaveformSum",  # charges over entire window
```

Comment:
Hum, I would have a concern here: the runs being processed for the linearity evaluation are typically flat-field runs acquired with different attenuation factors. They are thus typically non-pedestal runs, in the sense that pulsed events should dominate. The pedestal evaluation should thus, to my understanding, use another method than `FullWaveformSum`. 

I think I also raised the same question in the PR #239, see https://github.com/cta-observatory/nectarchain/pull/239#discussion_r2597737657

@tibaldo , could you please advise about that ?

Reply:
Hi @jlenain. I think the code should work correctly. The runs processed are flat-field with interleaved pedestal events. `PedestalNectarCAMCalibrationTool` uses only events of the type SKYPEDESTAL to estimate the pedestal.

If you are worried about a contamination from the flasher in interleaved pedestal events then I would recommend to use
```
filter_method="ChargeDistributionFilter",
charge_sigma_high_thr=3.0,
```
to exclude the tails of the charge distribution.


The subsequent call to the linearity tool does not use the charge in the output pedestal file but just the mean waveform values, so the integration method should be irrelevant unless you apply filtering as suggested above (which is not the case in the current version of the code with `filter_method=None`).